distro/src/avail/Tests.avail/Foundation Tests.avail/Math Tests.avail

```diff
@@ -627,7 +627,10 @@ Test "modular exponent tests" in math test suite is
 	Require: 110 ^ 45 mod 87 = 74;
 	Require: 1001 ^ 1001 mod 123 = 17;
 	Require: 30301829 ^ 1 mod (2^63) = 30301829;
-	Require: 12807421972 ^ 12345 mod 3201855493 = 0;
+	/* Super-cast it to whole number to avoid attempting to fold the leading
+	 * part as an invocation of "_^_".
+	 */
+	Require: (12807421972::whole number) ^ 12345 mod 3201855493 = 0;
```

Comment:
It was a big slowdown.  I sped up exponentiation in several ways to start to address this, including faster odd exponent check (the bitwise-and instruction), rewriting it to be iterative with loops that were handled by the optimizer, and multiplying the smallest numbers together first (to leverage int*int as long as possible).  Eventually, I introduced the supercast to avoid the folding entirely.  It would be fine to fold `"_^_mod_"`, since it's very fast, but unfortunately the current parser was also trying out `"_^_"`, producing a 120,000-digit integer in the process.